src/coreclr/jit/lclvars.cpp

```diff
@@ -4750,6 +4759,8 @@ void Compiler::lvaComputeRefCounts(bool isRecompute, bool setSlotNumbers)
         {
             varDsc->lvSingleDef             = varDsc->lvIsParam;
             varDsc->lvSingleDefRegCandidate = varDsc->lvIsParam;
+
+            varDsc->lvAllDefsAreNoGc = (varDsc->lvImplicitlyReferenced == false);
```

Comment:
```suggestion
            // if there are implicit refs, assume they may be gc defs
            varDsc->lvAllDefsAreNoGc = !varDsc->lvImplicitlyReferenced;
```

Reply:
This actually gives a compiler warning of "Using logical '!' when bitwise '~' was probably intended"